fili-generic-example/src/main/java/com/yahoo/wiki/webservice/application/GenericBinderFactory.java

```diff
@@ -0,0 +1,87 @@
+import com.yahoo.bard.webservice.druid.client.DruidWebService;
+import com.yahoo.bard.webservice.table.PhysicalTableDictionary;
+import com.yahoo.wiki.webservice.data.config.auto.DataSourceConfiguration;
+import com.yahoo.wiki.webservice.data.config.auto.DruidNavigator;
+import com.yahoo.wiki.webservice.data.config.dimension.GenericDimensions;
+import com.yahoo.wiki.webservice.data.config.metric.GenericMetricLoader;
+import com.yahoo.wiki.webservice.data.config.table.GenericTableLoader;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * Builds dimensions, metrics, and tables for all datasources found from druid.
+ */
+public class GenericBinderFactory extends AbstractBinderFactory {
+    private final Supplier<List<? extends DataSourceConfiguration>> configLoader;
+    private GenericDimensions genericDimensions;
+    private static Set<DimensionConfig> dimensions;
+
+    /**
+     * Constructs a GenericBinderFactory which starts configuring with druid.
+     */
+    public GenericBinderFactory() {
+        DruidWebService druidWebService = buildMetadataDruidWebService(getMappers().getMapper());
+        configLoader = new DruidNavigator(druidWebService);
+        configLoader.get();
+    }
+
+    /**
+     * Provides a way for {@link GenericMain} to access the dimensions.
+     * TODO: This is a bad solution to getting the dimensions from GenericMain
+     * @return the dimensions found from druid.
+     */
+    public static Set<DimensionConfig> getDimensions() {
+        return dimensions;
+    }
+
+    @Override
+    protected Set<DimensionConfig> getDimensionConfigurations() {
+        //NOTE: This is guaranteed to be called before getTableLoader()
+        genericDimensions = new GenericDimensions(configLoader);
+        dimensions = genericDimensions.getAllDimensionConfigurations();
+        return genericDimensions.getAllDimensionConfigurations();
+    }
+
+    @Override
+    protected TableLoader getTableLoader() {
+        return new GenericTableLoader(configLoader, genericDimensions, getDataSourceMetadataService());
+    }
+
+    @Override
+    protected DruidDimensionsLoader buildDruidDimensionsLoader(
+            DruidWebService webService,
+            PhysicalTableDictionary physicalTableDictionary,
+            DimensionDictionary dimensionDictionary
+    ) {
+        List<List<Dimension>> dimensionsList = genericDimensions.getAllDimensionConfigurations().stream()
+                .map(DimensionConfig::getApiName)
+                .map(dimensionDictionary::findByApiName)
+                .map(Collections::singletonList)
+                .collect(Collectors.toList());
+        return new DruidDimensionsLoader(
+                webService,
+                dimensionsList,
+                DruidDimensionsLoader.buildDataSourcesList(physicalTableDictionary)
```

Comment:
We should change it in the constructor so that if the `DRUID_DIM_LOADER_DIMENSIONS` is empty list, then we directly use everything in the dimensionDictionary, otherwise, use the `DRUID_DIM_LOADER_DIMENSIONS` list. This way we can just have one constructor without the static helper methods in `DruidDimensionLoader` class right?